### PR TITLE
Feature/update features comparison with ros2

### PR DESCRIPTION
**Pull request by FranFin**
Comments:

- New features introduced in the table.
- Updated/changed the description of some previous features.
- Set of symbols used updated and legend created below.
- Table split into sub-tables according to macro-categories defined in "Features and Architecture" section.
- Swapped the order of two elements in the "Features and Architecture" section so as to generate a better match with the new comparison table.

### CODE REVIEW

File: _docs/overview/ROS_2_feature_comparison/index.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ title: ROS 2 Feature Comparison
 permalink: /docs/overview/ROS_2_feature_comparison/
 ---
 
-Comparison of micro-ROS features with ROS 2 features. The following list has been compiled from [https://index.ros.org/doc/ros2/Features/](https://index.ros.org/doc/ros2/Features/) and [https://index.ros.org/doc/ros2/Roadmap/](https://index.ros.org/doc/ros2/Roadmap/).
+Comparison of micro-ROS features with ROS 2 features. The following list has been compiled from [https://index.ros.org/doc/ros2/Features/](https://index.ros.org/doc/ros2/Features/) and [https://index.ros.org/doc/ros2/Roadmap/](https://index.ros.org/doc/ros2/Roadmap/), and the features have been organized into sub-tables according to the macrocategories defined in the [Features and Architecture page](http://localhost:4000/docs/overview/features/).
```
Comment (pablogs9) [inline _docs/overview/ROS_2_feature_comparison/index.md:6]: careful with `localhost`
```suggestion
Comparison of micro-ROS features with ROS 2 features. The following list has been compiled from [https://index.ros.org/doc/ros2/Features/](https://index.ros.org/doc/ros2/Features/) and [https://index.ros.org/doc/ros2/Roadmap/](https://index.ros.org/doc/ros2/Roadmap/), and the features have been organized into sub-tables according to the macrocategories defined in the [Features and Architecture page](/overview/features/).
```

File: _docs/overview/ROS_2_feature_comparison/index.md [hunk 2]
```diff
@@ -12,34 +12,72 @@ Comparison of micro-ROS features with ROS 2 features. The following list has bee
   }
 </style>
 
+### Microcontroller-optimized client API supporting all major ROS concepts
+
+ROS 2 Feature | | Availability in micro-ROS
+-- | -- | --
+Common core client library that is wrapped by language-specific libraries | <span class="status_flag">&#10003;&#8728;</span> | Use of the client support library [rcl](https://github.com/ros2/rcl/) from ROS 2 as-is. The [rclc](https://github.com/micro-ROS/rclc) package provides convenience functions and an executor for use of rcl+rclc as client library for the C programming language. *Roadmap: migrate all functionalities to the rclc, so as to make it an independent abraction layer on top of the rcl that serves as user's API.*
```
Comment (pablogs9) [inline _docs/overview/ROS_2_feature_comparison/index.md:19]:
```suggestion
Common core client library that is wrapped by language-specific libraries | <span class="status_flag">&#10003;&#8728;</span> | Use of the client support library [rcl](https://github.com/ros2/rcl/) from ROS 2 as-is. The [rclc](https://github.com/micro-ROS/rclc) package provides convenience functions and an executor for use of RCL + RCLC as client library for the C programming language. *Roadmap: migrate all functionalities to the rclc, so as to make it an independent abraction layer on top of the rcl that serves as user's API.*
```

**Review by pablogs9 — CHANGES_REQUESTED** [review]

File: _docs/overview/ROS_2_feature_comparison/index.md (hunk 2 above)
Comment (pablogs9) [reply _docs/overview/ROS_2_feature_comparison/index.md:19]: I'm not sure, but if we decide to use a capitalized version, there are multiple places where there are `rclc` in lower case.

File: _docs/overview/ROS_2_feature_comparison/index.md (hunk 1 above)
Comment (ralph-lange) [reply _docs/overview/ROS_2_feature_comparison/index.md:6]: ... but the `/docs/` must be kept.

File: _docs/overview/ROS_2_feature_comparison/index.md (hunk 2 above)
Comment (ralph-lange) [reply _docs/overview/ROS_2_feature_comparison/index.md:19]: I always use lower case and write "Rcl" or "Rclc" only at the beginning of a sentence.

At index.ros.org, it seems to be written lower case in the majority of cases:
https://index.ros.org/search/?term=rcl&section=docs

And, rclcpp and rclpy are always written lower case:
https://index.ros.org/search/?term=rclcpp&section=docs

Comment (jamoralp) [inline _docs/overview/ROS_2_feature_comparison/index.md:19]:
```suggestion
Common core client library that is wrapped by language-specific libraries | <span class="status_flag">&#10003;&#8728;</span> | Use of the client support library [rcl](https://github.com/ros2/rcl/) from ROS 2 as-is. The [rclc](https://github.com/micro-ROS/rclc) package provides convenience functions and an executor for use of rcl+rclc as client library for the C programming language. *Roadmap: migrate all functionalities to the rclc, so as to make it an independent absraction layer on top of the rcl that serves as user's API.*
```

File: _docs/overview/ROS_2_feature_comparison/index.md [hunk 3]
```diff
@@ -12,34 +12,72 @@ Comparison of micro-ROS features with ROS 2 features. The following list has bee
...
+IDL | <span class="status_flag">&#10003;&#43;</span> | micro-ROS supports the same IDL types as ROS 2. Generation of C code from IDLs as handled by the Client is performed by the [Micro-XRCE-DDS-Gen](https://github.com/eProsima/Micro-XRCE-DDS-Gen) library, whereas generation of the C++ types handled by the Agent is handled by [Fast-DDS-Gen](https://github.com/eProsima/Fast-DDS-Gen).
+Logging | <span class="status_flag">&#8727;</span> | *Could be available as part of the standard logging mechanism in principle but not supported by Micro-XRCE-DDS due to dynamic message size. To be checked ...*
+
+### Multi-RTOS support with generic build system
+
+Feature | | Availability in micro-ROS
+-- | -- | --
+Build system | <span class="status_flag">&#10003;</span> | micro-ROS provides two ways of building a micro-ROS application. The first uses the [micro_ros_setup](https://github.com/micro-ROS/micro_ros_setup) tool integrated in a ROS 2 workspace. With this approach, the build systems of NuttX, FreeRTOS, and Zephyr are integrated with colcon. The other provides micro-ROS as a component of external development frameworks (e.g., ESP-IDF and Zephyr build system).
```
Comment (jamoralp) [inline _docs/overview/ROS_2_feature_comparison/index.md:57]:
```suggestion
Build system | <span class="status_flag">&#10003;</span> | micro-ROS provides two ways of building a micro-ROS application. The first uses the [micro_ros_setup](https://github.com/micro-ROS/micro_ros_setup) tool integrated in a ROS 2 workspace. With this approach, the build systems of NuttX, FreeRTOS, and Zephyr are integrated with colcon. The other provides micro-ROS as a component for external development frameworks (e.g., ESP-IDF and Zephyr build system).
```

File: _docs/overview/ROS_2_feature_comparison/index.md [hunk 4]
```diff
@@ -12,34 +12,72 @@ Comparison of micro-ROS features with ROS 2 features. The following list has bee
...
+Supported hardware | <span class="status_flag">&#10003;</span> | micro-ROS officially supports four boards. For the moments, all official ports are based on the STM32 series from ST and on the ESP32 from Espressif. Find more info [here](https://micro-ros.github.io/docs/overview/hardware/). More ports have been carried out by users, check the [complete list](https://github.com/micro-ROS/micro_ros_setup#supported-platforms).
```
Comment (jamoralp) [inline _docs/overview/ROS_2_feature_comparison/index.md:58]:
```suggestion
Supported hardware | <span class="status_flag">&#10003;</span> | micro-ROS officially supports four boards. For the moment, all official ports are based on the STM32 series from ST and on the ESP32 from Espressif. Find more info [here](https://micro-ros.github.io/docs/overview/hardware/). More ports have been carried out by users, check the [complete list](https://github.com/micro-ROS/micro_ros_setup#supported-platforms).
```

File: _docs/overview/ROS_2_feature_comparison/index.md [hunk 5]
```diff
@@ -12,34 +12,72 @@ Comparison of micro-ROS features with ROS 2 features. The following list has bee
...
+Continuous Integration | <span class="status_flag">&#10003;&#8728;</span> | Currently, the CI for micro-ROS is distributed to GitHub and GitLab. *Until the end of 2020, all CI should be moved migrated completely to the new CI actions of GitHub.* Please note that those packages that are released for standard ROS 2 are also built and tested on [build.ros2.org](http://build.ros2.org/).
+Documentation | <span class="status_flag">&#10003;</span> | High-level documentation at [micro-ros.github.io](https://micro-ros.github.io/). For detailed information please consult the README.md files in the relevant micro-ROS repositories at [github.com/micro-ROS/](https://github.com/micro-ROS/). For information on the middleware implementation, take a look at the [Micro XRCE-DDS documentation](https://micro-xrce-dds.docs.eprosima.com/en/latest/).
+Peer-to-peer functionality | <span class="status_flag">&#10003;&#8728;</span> | Prototypical peer-to-peer functionality implemented over boradcast. No QoS available for the moment. *Roadmap: improve prototype to achieve true point-to-point connection.*
```
Comment (jamoralp) [inline _docs/overview/ROS_2_feature_comparison/index.md:70]:
```suggestion
Peer-to-peer functionality | <span class="status_flag">&#10003;&#8728;</span> | Prototypical peer-to-peer functionality implemented over broadcast. No QoS available for the moment. *Roadmap: improve prototype to achieve true point-to-point connection.*
```

File: _docs/overview/ROS_2_feature_comparison/index.md (hunk 1 above)
Comment (pablogs9) [reply _docs/overview/ROS_2_feature_comparison/index.md:6]: Suggested fix fixed

File: _docs/overview/ROS_2_feature_comparison/index.md (hunk 2 above)
Comment (pablogs9) [reply _docs/overview/ROS_2_feature_comparison/index.md:19]: Resolving conversation, I'm ok with lowercase

**Review by ralph-lange — DISMISSED** [review]

**Review by ralph-lange — APPROVED** [review]

**Review by jamoralp — APPROVED** [review]

**Review by pablogs9 — APPROVED** [review]

**Pull request merged**